src/tsf/Implementation.cpp

```diff
@@ -84,8 +84,20 @@ void Implementation::Uninitialize() noexcept
     }
 }
 
-HWND Implementation::FindWindowOfActiveTSF() const noexcept
-{
+HWND Implementation::FindWindowOfActiveTSF() noexcept
+{
+    // We don't know what ITfContextOwner we're going to get in
+    // the code below and it may very well be us (this instance).
+    // It's also possible that our IDataProvider's GetHwnd()
+    // implementation calls this FindWindowOfActiveTSF() function.
+    // This can then in infinite recursion because we're calling
+    // GetWnd() below, which may call GetHwnd(), which may call
+    // FindWindowOfActiveTSF(), and so on.
+    // By temporarily clearing the _provider we fix that flaw.

```

Comment:
Where is the determination to bail out made based on `_provider`?

Reply:
Throughout the file. All places that use `_provider` are already doing a null check.